designer/preprocessing/util.py

```diff
@@ -381,37 +416,38 @@ def cat(self, path, ext='.nii' ,verbose=False, force=False):
         force:      bool
             Forces file overwrite if they already exist
         """
-        if self.nDWI <= 1:
-            raise Exception('Nothing to concatenate when there is '
-        'only one input series.')
-        else:
-            miflist = []
-            # The following loop first converts NifTis to a .mif file
-            for (idx, i) in enumerate(self.DWIlist):
+        # if self.nDWI <= 1:
+        #     raise Exception('Nothing to concatenate when there is '
+        # 'only one input series.')
+        miflist = []
+        # The following loop first converts NifTis to a .mif file
+        for (idx, i) in enumerate(self.DWIlist):
+            if 'nifti' in self.InputType:
                 self.json2fslgrad(i)
-                convert_args = ['mrconvert -stride 1,2,3,4']
-                if verbose is False:
-                    convert_args.append('-quiet')
-                if force is True:
-                    convert_args.append('-force')
-                if op.exists(self.BVEClist[idx]) or \
-                        op.exists(self.BVALlist[idx]):
-                    convert_args.append('-fslgrad')
-                    convert_args.append(self.BVEClist[idx])
-                    convert_args.append(self.BVALlist[idx])
-                if op.exists(self.JSONlist[idx]):
-                    convert_args.append('-json_import')
-                    convert_args.append(self.JSONlist[idx])
-                convert_args.append(i)
-                convert_args.append(
-                    op.join(path, ('dwi' + str(idx) + '.mif')))
-                miflist.append(op.join(path, ('dwi' + str(idx) + '.mif')))
-                cmd = ' '.join(str(e) for e in convert_args)
-                completion = subprocess.run(cmd, shell=True)
-                if completion.returncode != 0:
-                    raise Exception('Conversion to .mif failed.')
-            # The following command concatenates all DWI(i) in a single
-            # .mif file
+            convert_args = ['mrconvert -stride 1,2,3,4']
+            if verbose is False:
+                convert_args.append('-quiet')
+            if force is True:
+                convert_args.append('-force')
+            if op.exists(self.BVEClist[idx]) or \
```

Comment:
Wouldn't you perform these checks only if the input is a .nii or a .nii.gz?

Reply:
It will form the BVEC/BVAL/JSON extensions regardless of input type, but will use them iff input type is nifti or compressed nifti. It was a lot easier to control processing flow than checking memory state of variables and forming separate/additional if-else statements.

Reply:
Makes sense.